lib/block_view_types.go

```diff
@@ -1102,6 +1306,84 @@ func IsVanillaRepost(postEntry *PostEntry) bool {
 	return !postEntry.IsQuotedRepost && postEntry.RepostedPostHash != nil
 }
 
+func (pe *PostEntry) Encode() []byte {
+	var data []byte
+
+	data = append(data, EncodeByteArray(pe.PostHash.ToBytes())...)
+	data = append(data, EncodeByteArray(pe.PosterPublicKey)...)
+	data = append(data, EncodeByteArray(pe.ParentStakeID)...)
+	data = append(data, EncodeByteArray(pe.Body)...)
+	if pe.RepostedPostHash != nil {
+		data = append(data, EncodeByteArray(pe.RepostedPostHash.ToBytes())...)
+	} else {
+		data = append(data, UintToBuf(0)...)
+	}
```

Comment:
Yeah I made an update to encoders with adding the interface, as I mentioned in the other comment. Byte arrays are chill and even if they're empty, they will be encoded correctly, that is they will have len() = 0. Pointers are a bit more tricky because if they're nil they would most likely fail to be encoded. That's why I made a standard to first encode a "existence byte" which is a single byte that tells you if a pointer was nil or not.